OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -162,29 +164,19 @@ protected virtual void Tick(Actor self)
 
 		void INotifyIdle.TickIdle(Actor self)
 		{
-			if (state != AnimationState.Idle && state != AnimationState.IdleAnimating && state != AnimationState.Attacking)
+			if (!AllowIdleAnimation(self))
+				return;
+
+			if (state == AnimationState.Waiting && hasIdleSequence)
```

Comment:
If `AllowIdleAnimation` already checks for `hasIdleSequence`, I don't think we have to check here again. (I.e. `hasIdleSequence` can never be `false` here.)